testsuite/libra-fuzzer/src/fuzz_targets/safety_rules.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{corpus_from_strategy, fuzz_data_to_value, FuzzTargetImpl};
+use libra_proptest_helpers::ValueGenerator;
+use safety_rules::{fuzzing_utils::arb_handle_message_input, test_utils};
+
+#[derive(Clone, Debug, Default)]
+pub struct SafetyRulesHandleMessage;
+
+/// This implementation will fuzz the handle_message() method of the safety rules serializer
+/// service.
+impl FuzzTargetImpl for SafetyRulesHandleMessage {
+    fn description(&self) -> &'static str {
+        "Safety rules: handle_message()"
+    }
+
+    fn generate(&self, _idx: usize, _gen: &mut ValueGenerator) -> Option<Vec<u8>> {
+        Some(corpus_from_strategy(arb_handle_message_input()))
+    }
+
+    fn fuzz(&self, data: &[u8]) {
+        // Generate the fuzzer input
+        let (safety_rules_input, verify_vote_proposal_sig, signer_byte) =
+            fuzz_data_to_value(data, arb_handle_message_input());
+
+        // Create a safety rules serializer test instance for fuzzing
+        let mut serializer_service =
```

Comment:
@mimoo, one of the disadvantages of fuzzing a method (as opposed to a function), is that we need to create a new   object before running each fuzz iteration. 
1. I'm guessing there's no easy way around this without completely refactoring all the code (which may not be possible to remove dependencies)?
2. Alternatively, do you know if it's possible to create a single object that is used by all fuzzing iterations? Although, the disadvantages of this approach is that each fuzz iteration may cause the object to enter a state that that prevents us from exploring all states/paths.

Reply:
Yeah, true, but if I'm re-initializing it before each run, it's effectively the same logical behaviour, right? In that case, I'll just leave the object creation here and avoid exposing a global.

Reply:
depending on the situation sometimes it's faster to re-initialize an already created object : o